### PR TITLE
Adds native binary support for Darwin Arm64 (e.g. M1 Mac)

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Adds native binary support for Darwin Arm64 (e.g. M1 Mac)

### CODE REVIEW

**Comment by berndverst** [comment]
This install script change works:

```
./install/install.sh
Your system is darwin_arm64
Installing Dapr CLI...

Getting the latest Dapr CLI...
Installing v1.4.0 Dapr CLI...
Downloading https://github.com/dapr/cli/releases/download/v1.4.0/dapr_darwin_arm64.tar.gz
```

(That of course fails because this artifact doesn't exist, but it will start working with a later 1.5.0 release candidate and the 1.5.0 release)

**Review by pkedy — DISMISSED** [review]

File: install/install.sh [hunk 1]
```diff
@@ -50,14 +50,6 @@ verifySupported() {
         fi
     done
 
-    if [ "$current_osarch" == "darwin-arm64" ]; then
```
Comment (wcs1only) [inline install/install.sh:53]: So we need a bit more robust fallback than this. The install.sh script is unversioned and live, which means this change will break M1 macs until such time as the release is published. I think what we need to do is "if darwin-arm64 && darwin/arm64 exists and is public then download it else amd64"

Comment (berndverst) [reply install/install.sh:53]: Makes sense -- keep in mind this script is local to the 1.5 branch right now, but I agree we should have this.

Comment (wcs1only) [reply install/install.sh:53]: > Makes sense -- keep in mind this script is local to the 1.5 branch right now, but I agree we should have this.

True, but this also means it won't work until we merge release branch back to master, and thus we'll have no way to test it works before we go live.

**Comment by berndverst** [comment]
After this change behavior is as before where Darwin Arm64 binary does not exist:
```
./install.sh
The darwin_arm64 arch has no native binary for this version of Dapr, however you can use the amd64 version so long as you have rosetta installed
Use 'softwareupdate --install-rosetta' to install rosetta if you don't already have it
Installing Dapr CLI...

Getting the latest Dapr CLI...
Installing v1.4.0 Dapr CLI...
Downloading https://github.com/dapr/cli/releases/download/v1.4.0/dapr_darwin_amd64.tar.gz ...
```

**Review by wcs1only — APPROVED** [review]
LGTM

**Pull request merged**